src/coreComponents/dataRepository/BufferOps_inline.hpp

```diff
@@ -1519,16 +1530,68 @@ Unpack( buffer_unit_type const * & buffer,
       }
     }
 
-    // insert known local indices into the set of indices
-    //  related to the local index
-    localIndex const numUniqueMapped = LvArray::sortedArrayManipulation::makeSortedUnique( mapped.begin(), mapped.end() );
-    var.insertIntoSet( li, mapped.begin(), mapped.begin() + numUniqueMapped );
+    // insert known local indices into the set of indices related to the local index
+    mapped.resize( LvArray::sortedArrayManipulation::makeSortedUnique( mapped.begin(), mapped.end() ) );
+    std::set_difference( mapped.begin(), mapped.end(), var[li].begin(), var[li].end(), std::back_inserter( mappedNew ) );
+    localIndex const numNewValues = LvArray::integerConversion< localIndex >( mappedNew.size() );
+
+    // check if we have enough capacity to insert new indices
+    if( numNewValues <= var.capacityOfSet( li ) - var.sizeOfSet( li ) )
+    {
+      // all good, just insert new entries
+      var.insertIntoSet( li, mappedNew.begin(), mappedNew.end() );
+    }
+    else
+    {
+      // need to stash away and rebuild the map later
+      localIndex const k = indiciesToInsert.size();
+      indiciesToInsert.emplace_back( li );
+      valuesToInsert.appendSet( numNewValues );
+      valuesToInsert.insertIntoSet( k, mappedNew.begin(), mappedNew.end() );
+    }
 
-    // insert unknown global indices related to the local index
-    //  into an additional mapping to resolve externally
-    localIndex const numUniqueUnmapped = LvArray::sortedArrayManipulation::makeSortedUnique( unmapped.begin(), unmapped.end() );
-    unmappedGlobalIndices[li].insert( unmapped.begin(), unmapped.begin() + numUniqueUnmapped );
+    // insert unknown global indices related to the local index into an additional mapping to resolve externally
+    unmapped.resize( LvArray::sortedArrayManipulation::makeSortedUnique( unmapped.begin(), unmapped.end() ) );
+    unmappedGlobalIndices[li].insert( unmapped.begin(), unmapped.end() );
   }
+
+  // If there were element lists that didn't fit in the map, rebuild the whole thing
+  if( !indiciesToInsert.empty() )
+  {
+    // Copy old capacities (no way to direct copy, must kernel launch)
```

Comment:
Oof. Definitely feels like this is something we should have a util function for in lvarray.

Reply:
Yeah, since capacities are not stored directly but rather as a running sum, minor computation is required to get them out, so a function in `LvArray` would have to take a RAJA execution policy and I just didn't feel like adding something like this. On the other hand, it's a counterpart to `resizeFromCapacities` which does take a policy, so maybe it's ok.